apps/src/dance/Dance.js

```diff
@@ -421,6 +422,11 @@ Dance.prototype.afterInject_ = function () {
 };
 
 Dance.prototype.playSong = function (url, callback, onEnded) {
+  if (Sounds.getSingleton().soundsById[url].isPlaying()) {
```

Comment:
Can we just use `Sounds.getSingleton().isPlaying(url)` for this? Do we need to check `soundsById` specifically?